canonical-web-and-design/maas-ui
# fix(legacy): update dnsresource on name change

## Done
Resolve issue that resulted in a serverside crash when updating the name of a dns resource.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

* Create DNS record with name: "test-internal", type "A", value "1.2.3.4"
* Rename it to "test"
* Try to rename it back to "test-internal"
* Ensure the rename is successful, and the server does not return `list.remove(x): x not in list message will appear in UI`

* Rename a non A record (e.g. TXT)
* Ensure the rename is successful

## Fixes

Fixes: #1728 

## Launchpad issue

lp#1894894

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
